diff --git a/Dockerfile.s390x b/Dockerfile.s390x
index afd20ec..c226246 100644
--- a/Dockerfile.s390x
+++ b/Dockerfile.s390x
@@ -1,4 +1,4 @@
-# Copyright (c) 2017-2018 Tigera, Inc. All rights reserved.
+# Copyright (c) 2018-2019 Tigera, Inc. All rights reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -11,8 +11,43 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+ARG GIT_VERSION=unknown
+
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest as build
+RUN mkdir -p /tmp/dikastes
+RUN chmod 0777 /tmp/dikastes
+
+# Add in top-level license file
+RUN mkdir /licenses
+COPY LICENSE /licenses
+
 FROM scratch
+ARG GIT_VERSION
+LABEL name="Calico Dikastes" \
+      vendor="Project Calico" \
+      version=$GIT_VERSION \
+      release="1" \
+      summary="Calico Dikastes enables Application Layer Policy" \
+      description="Calico Dikastes enables Application Layer Policy" \
+      maintainer="Laurence Man<laurence@tigera.io>"
 
+COPY --from=build /licenses /licenses
 ADD bin/dikastes-s390x /dikastes
 ADD bin/healthz-s390x /healthz
-CMD ["/dikastes"]
+
+# Include libraries from UBI for dynamic linking.
+COPY --from=build /lib64 /lib64
+COPY --from=build /lib /lib
+
+# Typical Linux systems start numbering human users at 1000, reserving 1-999
+# for services, so we pick 999 to be least likely to overlap.  It's not a big
+# deal if we happen to overlap, as it would take a container escape for
+# an attacker to use any privileges that happen to be assigned to this user.
+#
+# Precreate the /var/run/dikastes directory so that we don't need any elevated
+# permission to create the directory at runtime.
+COPY --chown=999 --from=build /tmp/dikastes /var/run/dikastes
+USER 999
+ENTRYPOINT ["/dikastes"]
+CMD ["server"]
diff --git a/Makefile b/Makefile
index a2f11d6..fb396ec 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 PACKAGE_NAME?=github.com/projectcalico/app-policy
-GO_BUILD_VER?=v0.58
+GO_BUILD_VER?=v0.59
 
 ORGANIZATION=projectcalico
 SEMAPHORE_PROJECT_ID?=$(SEMAPHORE_APP_POLICY_PROJECT_ID)
@@ -158,14 +158,14 @@ proto/felixbackend.pb.go: proto/felixbackend.proto
 		      $(PROTOC_CONTAINER) \
 		      $(PROTOC_IMPORTS) \
 		      proto/*.proto \
-		      --gogofast_out=plugins=grpc:proto
+		      --gogofaster_out=plugins=grpc:proto
 
 proto/healthz.pb.go: proto/healthz.proto
 	$(DOCKER_RUN) -v $(CURDIR):/src:rw \
 		      $(PROTOC_CONTAINER) \
 		      $(PROTOC_IMPORTS) \
 		      proto/*.proto \
-		      --gogofast_out=plugins=grpc:proto
+		      --gogofaster_out=plugins=grpc:proto
 
 
 # Building the image
