diff --git a/Dockerfile.s390x b/Dockerfile.s390x
index 5daed33..78fab27 100644
--- a/Dockerfile.s390x
+++ b/Dockerfile.s390x
@@ -11,9 +11,36 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+ARG GIT_VERSION=unknown
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest as base
+
+# Add in top-level license file
+RUN mkdir /licenses
+COPY LICENSE /licenses
+
+# Make sure the status file is owned by our user.
+RUN mkdir /status
+RUN touch /status/status.json && chown 999 /status/status.json
+
 FROM scratch
-LABEL maintainer "LoZ Open Source Ecosystem (https://www.ibm.com/developerworks/community/groups/community/lozopensource)"
+ARG GIT_VERSION
+LABEL name="Calico Kubernetes controllers" \
+      vendor="Project Calico" \
+      version=$GIT_VERSION \
+      release="1" \
+      summary="Calico Kubernetes controllers monitor the Kubernetes API and perform actions based on cluster state" \
+      description="Calico Kubernetes controllers monitor the Kubernetes API and perform actions based on cluster state" \
+      maintainer="Casey Davenport <casey@tigera.io>"
+
+COPY --from=base /licenses /licenses
+COPY --from=base /status /status
+
+COPY --from=base /usr/include /usr/include
+COPY --from=base /lib64 /lib64
+COPY --from=base /lib /lib
 
 ADD bin/kube-controllers-linux-s390x /usr/bin/kube-controllers
 ADD bin/check-status-linux-s390x /usr/bin/check-status
+USER 999
 ENTRYPOINT ["/usr/bin/kube-controllers"]
diff --git a/docker-images/flannel-migration/Dockerfile.s390x b/docker-images/flannel-migration/Dockerfile.s390x
index 1d2bc22..a21f859 100644
--- a/docker-images/flannel-migration/Dockerfile.s390x
+++ b/docker-images/flannel-migration/Dockerfile.s390x
@@ -11,10 +11,35 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+ARG GIT_VERSION=unknown
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest as base
+
+# Add in top-level license file
+RUN mkdir /licenses
+COPY LICENSE /licenses
+
+# Make sure the status file is owned by our user.
+RUN mkdir /status
+RUN touch /status/status.json && chown 999 /status/status.json
+
 FROM scratch
-LABEL maintainer "Song Jiang <song@tigera.io>"
+LABEL name="Calico Flannel migration controller" \
+      vendor="Project Calico" \
+      version=$GIT_VERSION \
+      release="1" \
+      summary="Calico Flannel migration controller updates a flannel cluster to Calico" \
+      description="Calico Flannel migration controller updates a flannel cluster to Calico" \
+      maintainer="Song Jiang <song@tigera.io>"
+
+COPY --from=base /licenses /licenses
+COPY --from=base /status /status
+COPY --from=base /usr/include /usr/include
+COPY --from=base /lib64 /lib64
+COPY --from=base /lib /lib
 
 ADD bin/kubectl-s390x /usr/bin/kubectl
 ADD bin/kube-controllers-linux-s390x /usr/bin/kube-controllers
 ADD bin/check-status-linux-s390x /usr/bin/check-status
+USER 999
 ENTRYPOINT ["/usr/bin/kube-controllers"]
